### PR TITLE
Implement initial bash CNB template

### PR DESCRIPTION
**Pull request by AidanDelaney**
* Provide scaffolding for Go and bash buildpacks

Scaffold Go and bash buildpacks in the CNB style

* Reflect latest RFC changes in README

Ensure that the UI presented in the README is the same as the proposed
UI from the RFC.

### CODE REVIEW

File: Go/{{.ProjectDirectory}}/go.mod [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module {{ .ModuleName }}
+
+go 1.18
+
+require (
+	github.com/BurntSushi/toml v1.1.0 // indirect
+	github.com/Masterminds/semver/v3 v3.1.1 // indirect
+	github.com/buildpacks/libcnb v1.26.0 // indirect
+	github.com/onsi/gomega v1.19.0 // indirect
+)
```
Comment (dmikusa) [inline Go/{{.ProjectDirectory}}/go.mod]: I wonder if dependabot will keep these up to date or not? 🤔

Regardless, we may want the template to create a README.md or TODO.md in the generated buildpack directory with some next steps for the user. Updating go modules could be one thing. Modify build and detect or create a new module and tests for specific functionality.

**Review by dmikusa — COMMENTED** [review]
Where did we land on generating test files for build and detect?

Also, what about a Makefile? Or is that getting too opinionated?

File: Go/{{.ProjectDirectory}}/go.mod (hunk 1 above)
Comment (AidanDelaney) [reply Go/{{.ProjectDirectory}}/go.mod]: I've added a `README.md`.  I don't think dependabot will keep our dependencies up-to-date as the `go.mod` file does not parse.  It does not parse due to the `module {{ .ModuleName }}` template variable.

**Comment by dmikusa** [comment]
> Where did we land on generating test files for build and detect?
> 
> Also, what about a Makefile? Or is that getting too opinionated?

https://github.com/buildpacks/profile/blob/main/profile/detect_test.go and https://github.com/buildpacks/profile/blob/main/profile/build_test.go for some examples of tests, and a Makefile https://github.com/buildpacks/profile/blob/main/Makefile.

**Comment by AidanDelaney** [comment]
I like the suggested tests. I'll factor out the Go implementation from this PR so that we can give it better consideration.

**Review by dmikusa — APPROVED** [review]

**Pull request merged**